subsys/bluetooth/mesh/provisioner.c

```diff
@@ -589,18 +589,16 @@ static void prov_node_add(void)
 	struct bt_mesh_cdb_node *node = provisionee.node;
 	int err;
 
-	if (atomic_test_bit(bt_mesh_prov_link.flags, REPROVISION)) {
-		bt_mesh_cdb_node_update(node, bt_mesh_prov_link.addr,
-					provisionee.elem_count);
-	}
-
 	err = bt_mesh_cdb_node_key_import(node, provisionee.new_dev_key);
 	if (err) {
 		LOG_ERR("Failed to import node device key");
 		return;
 	}
 
-	if (IS_ENABLED(CONFIG_BT_SETTINGS)) {
+	if (atomic_test_bit(bt_mesh_prov_link.flags, REPROVISION)) {
```

Comment:
~~Shouldn't this also be gated by IS_ENABLED(settings) (or just add single bigger outer if with IS_ENABLED(settings))?~~
```suggestion
	if (IS_ENABLED(CONFIG_BT_SETTINGS) && atomic_test_bit(bt_mesh_prov_link.flags, REPROVISION)) {
```

Now I see, we don't have consistency in CDB API. Some APIs check `IS_ENABLED(settings)` internally, some don't.